sdk/core/core/inc/az_http_response_parser.h

```diff
@@ -61,14 +49,28 @@ typedef struct {
  * Initializes an HTTP response parser.
  */
 AZ_NODISCARD az_result
-az_http_response_parser_init(az_span const buffer, az_http_response_parser * const out);
+az_http_response_parser_init(az_http_response_parser * const out, az_span const buffer);
+
+/**
+ * An HTTP status line.
+ */
+AZ_NODISCARD az_result az_http_response_parser_get_status_line(
+    az_http_response_parser * const self,
+    az_http_response_status_line * const out);
+
+/**
+ * An HTTP header.
+ */
+AZ_NODISCARD az_result az_http_response_parser_get_header(
+    az_http_response_parser * const self,
+    az_http_response_header * const out);
 
 /**
- * Returns the next HTTP response value.
+ * An HTTP body.
  */
-AZ_NODISCARD az_result az_http_response_parser_read(
+AZ_NODISCARD az_result az_http_response_parser_get_body(
```

Comment:
This seems like it would be inefficient since it would have to parse through all the headers to get to the body.
My design had you passing in the lastheader in order to get the body.
Your design allows a call to this method before any calls to az_http_response_parser_get_next_header.

Reply:
It the same. The function works in `O(1)` because the first parameter is a parser which maintains the state.